force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -228,18 +298,39 @@ public with sharing class ClassesObjectsMethods {
     public class Book {
         // Declare three public instance variables - title, author, pages
         // title
+        public String title;
         // author
+        public String author;
         // pages
+        public Integer pages;
     
         // Book constructor to initialize the title and author
-        public Book() {
-            // Initialize the title and author
+        public Book(String title, String author, Integer pages) {
```

Comment:
You added an extra parameter here for pages that you didn't need to. That is why you had to adjust the test classes to make it work. You could also have have overloaded the constructor (made 2 versions). One with 2 inputs for title and author, and 1 with 3 for title, author, and pages.